evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -183,12 +183,18 @@ def predict(self, X, objective=None, X_train=None, y_train=None):
             y_train (pd.Series or None): Training labels.
```

Comment:
@ParthivNaresh if we need `X_train` and `y_train` should we still accept `None` as input? I don't have the context on why we allowed this before!

Reply:
Yeah I'm not sure why `X_train` and `y_train` are optional here, they're very much required

Reply:
Oh I think it had to do with the fact that the parent `PipelineBase` has them as optional because regular regression and classification problems don't need them